src/core/iBlockchain.ts

```
@@ -0,0 +1,42 @@
+        transaction: Transaction
+    ): Result<number, CoreError>
+    hashBlock(
+        previousBlockHash: string,
+        currentBlockData: BlockData,
+        nonce: number
+    ): string
+    proofOfWork(
+        previousBlockHash: string,
+        currentBlockData: BlockData
+    ): Result<number, CoreError>
+    chainIsValid(blockchain: Block[]): Result<{}, CoreError>
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)